scrapy_zyte_api/_params.py

```diff
@@ -629,32 +726,21 @@ def parse(self, request):
             cookies_enabled=cookies_enabled,
             cookie_jars=self._cookie_jars,
             max_cookies=self._max_cookies,
+            experimental_cookies=self._experimental_cookies,
         )
-        if not dont_merge_cookies and self._warn_on_cookies:
-            self._handle_warn_on_cookies(request, params)
+        if (
+            params
+            and self._unreported_deprecated_experimental_fields
+            and "experimental" in params
+        ):
+            for field in list(self._unreported_deprecated_experimental_fields):
+                if field in params["experimental"]:
+                    self._unreported_deprecated_experimental_fields.remove(field)
+                    logger.warning(
+                        f"Zyte API parameters for request {request} include "
+                        f"experimental.{field}, which is deprecated. Please, "
+                        f"replace it with {field}, both in request parameters "
+                        f"and in any response parsing logic that might rely "
+                        f"on the old parameter."
+                    )
```

Comment:
I went with logging warning messages instead of using `warnings.warn` with `DeprecationWarning` mostly because it made it easier to use tests (which already had warning log support). I also wonder if logging a warning might be better for Scrapy Cloud support, or other cloud systems. But no strong opinion.